src/components/Table/GridTable.tsx

```diff
@@ -272,7 +265,11 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     resizeTarget,
   } = props;
 
-  const [collapsedIds, collapseAllContext, collapseRowContext] = useToggleIds(rows, persistCollapse);
+  // Create a ref that always contains the latest rows, for our effectively-singleton RowState to use
+  const rowsRef = useRef(rows);
+  rowsRef.current = rows;
+  const [rowState] = useState(() => new RowState(rowsRef, persistCollapse));
```

Comment:
We'll have only 1 `RowState` instance for the lifetime of GridTable, and the various components (i.e. `Collapse`) that render based on it, or mutate it, will get it via a context.